openpayprestashop.php

```diff
@@ -24,985 +24,1068 @@
+                        'country_code' => 'MX'
+                    )
+                );
+
+                $customer_openpay = $this->createOpenpayCustomer($customer_data);
+
+                Db::getInstance()->Execute('
                         INSERT INTO '._DB_PREFIX_.'openpay_customer (id_openpay_customer, openpay_customer_id, id_customer, date_add)
-                        VALUES (null, \''.pSQL($customer_openpay->id).'\', '.(int)$this->context->cookie->id_customer.', NOW())');
-
-				return $customer_openpay;
-			}
-			catch (Exception $e)
-			{
-				if (class_exists('Logger'))
-					Logger::addLog($this->l('Openpay - Can not create Openpay Customer'), 1, null, 'Cart', (int)$this->context->cart->id, true);
-			}
-		}
-		else
-			return $this->getCustomer($openpay_customer['openpay_customer_id']);
-	}
-
-	public function getCustomer($customer_id)
-	{
-		$pk = Configuration::get('OPENPAY_MODE') ? Configuration::get('OPENPAY_PRIVATE_KEY_LIVE') : Configuration::get('OPENPAY_PRIVATE_KEY_TEST');
-		$id = Configuration::get('OPENPAY_MODE') ? Configuration::get('OPENPAY_MERCHANT_ID_LIVE') : Configuration::get('OPENPAY_MERCHANT_ID_TEST');
-
-		$openpay = Openpay::getInstance($id, $pk);
-		Openpay::setProductionMode(Configuration::get('OPENPAY_MODE'));
-
-		$customer = $openpay->customers->get($customer_id);
-		return $customer;
-	}
-
-	public function createOpenpayCustomer($customer_data)
-	{
-		$pk = Configuration::get('OPENPAY_MODE') ? Configuration::get('OPENPAY_PRIVATE_KEY_LIVE') : Configuration::get('OPENPAY_PRIVATE_KEY_TEST');
-		$id = Configuration::get('OPENPAY_MODE') ? Configuration::get('OPENPAY_MERCHANT_ID_LIVE') : Configuration::get('OPENPAY_MERCHANT_ID_TEST');
-
-		$openpay = Openpay::getInstance($id, $pk);
-		Openpay::setProductionMode(Configuration::get('OPENPAY_MODE'));
-
-		try
-		{
-			$customer = $openpay->customers->add($customer_data);
-			return $customer;
-		}
-		catch (Exception $e)
-		{
-			$this->error($e);
-		}
-	}
-
-	public function createOpenpayCharge($customer, $charge_request)
-	{
-		$pk = Configuration::get('OPENPAY_MODE') ? Configuration::get('OPENPAY_PRIVATE_KEY_LIVE') : Configuration::get('OPENPAY_PRIVATE_KEY_TEST');
-		$id = Configuration::get('OPENPAY_MODE') ? Configuration::get('OPENPAY_MERCHANT_ID_LIVE') : Configuration::get('OPENPAY_MERCHANT_ID_TEST');
-
-		Openpay::getInstance($id, $pk);
-		Openpay::setProductionMode(Configuration::get('OPENPAY_MODE'));
-
-		try
-		{
-			$charge = $customer->charges->create($charge_request);
-			return $charge;
-		}
-		catch (Exception $e)
-		{
-			$this->error($e);
-		}
-	}
-
-	public function createWebhook()
-	{
-		$domain = (Configuration::get('PS_SSL_ENABLED') ? 'https' : 'http').'://'.$_SERVER['HTTP_HOST'];
-		$webhook_data = array(
-			'url' => $domain.__PS_BASE_URI__.'modules/openpayprestashop/notification.php',
-			'user' => Configuration::get('OPENPAY_WEBHOOK_USER'),
-			'password' => Configuration::get('OPENPAY_WEBHOOK_PASSWORD'),
-			'event_types' => ar
```

Comment:
It's looks like if Openpay customer wasn't created.
Can i ask you a favor, could you take a screenshot of your "login Openpay account? Just to corroborate this theory and check why wasn't created.

Thanks!